cpp/src/kmeans/kmeans_mg_impl.cuh

```diff
@@ -620,6 +657,9 @@ void fit(const raft::handle_t &handle, const KMeansParams &params,
   MLCommon::device_buffer<char> workspace(handle.get_device_allocator(),
                                           stream);
 
+  // check if weights sum up to n_samples
+  checkWeights(handle, workspace, weight, stream);
```

Comment:
Great you pulled this over from cumlprims! IIRC, the one remaining issue should be that the single-GPU k-means normalizes the weights in predict, however that will cause the multi-gpu version to normalize each partition individually since it's embarrassingly parallel.

The weights are being normalized globally in the Dask-based predict but the single-GPU predict is going to re-normalize them locally. The more straightforward path to fixing this might be to have the C++ `predict()` function accept a `normalize_weights` argument which defaults to `true` but we can have the multi-GPU predict function flip it off. The goal here is to eliminate the need for `predict()` to use the comms, because then it would no longer be able to execute embarassingly parallel.